### PR TITLE
Allow for schema addition without default tree during bootstrap

### PR DESCRIPTION
**Pull request by KoltesDigital**
**Description of the change**

It is currently not possible to skip default tree generation while having default schemas. While these default schemas could be added in the custom LDIF directory, they are also part of OpenLDAP, i.e. they are already embedded in the Docker image. If someday the defaults change (see #26), or less likely the files change, I'd like that they also change for me, which won't happen if I had to copy them in a own directory.

Well, the patch just inverts two nested tests.

**Benefits**

Having default schemas with a custom organization tree.

**Possible drawbacks**

Configuration with `LDAP_SKIP_DEFAULT_TREE=yes` and an empty `LDAP_CUSTOM_LDIF_DIR` had only the core schema before this patch, and will have extra schemas (cosine, inetorgperson, nis) thereafter. IMHO `LDAP_SKIP_DEFAULT_TREE=yes` should be used when `LDAP_CUSTOM_LDIF_DIR` is not empty, so the case should be rare.

**Applicable issues**

None

**Additional information**

Not important: I think this comment adds little value. https://github.com/bitnami/bitnami-docker-openldap/blob/master/2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh#L382

### CODE REVIEW

File: 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh [hunk 1]
```diff
@@ -376,13 +376,13 @@ ldap_initialize() {
...
-            info "Skipping default schemas/tree structure"
+        if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
+            ldap_add_custom_ldifs
         else
             # Initialize OpenLDAP with schemas/tree structure
             ldap_add_schemas
-            if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
-                ldap_add_custom_ldifs
+            if is_boolean_yes "$LDAP_SKIP_DEFAULT_TREE"; then
+                info "Skipping default schemas/tree structure"
```
Comment (andresbono) [inline 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:385]: Should this message be changed?
```suggestion
                info "Skipping default tree structure"
```

File: 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh [hunk 2]
```diff
@@ -376,13 +376,13 @@ ldap_initialize() {
         if is_boolean_yes "$LDAP_ENABLE_TLS"; then
             ldap_configure_tls
         fi
-        if is_boolean_yes "$LDAP_SKIP_DEFAULT_TREE"; then
-            info "Skipping default schemas/tree structure"
+        if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
+            ldap_add_custom_ldifs
```
Comment (andresbono) [inline 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:380]: When entering this block of the if, should we show a message saying that we are going to skip default schemas/tree structure?

**Review by andresbono — COMMENTED** [review]
Can you also modify the documentation for the environment variables in the README so that this new behavior is explained?

For instance, you can say that when providing `LDAP_CUSTOM_LDIF_DIR`, schemas and tree generation won't happen.

**Comment by andresbono** [comment]
Thanks for your contribution!! Could you please check my comments?

File: 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh (hunk 2 above)
Comment (KoltesDigital) [reply 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:380]: Note that there's already a warning in that function, but I'm changing the patch. Overall it seems that all functions which are called from `ldap_initialize` issue their own messages, so this function only issue messages for empty blocks.

File: 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh (hunk 1 above)
Comment (KoltesDigital) [reply 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:385]: Agreed.

**Comment by KoltesDigital** [comment]
Thanks for the review! I've changed the branch (as the patch is small, I amended the commit to avoid cluttering the history).

I'm now proposing to promote "default tree" as a vocabulary term (think "set expression") in the documentation and code. Thus I've added a small section in the README to explain what this default tree is about (moving some info from the configuration section). I think this makes things a bit clearer.

File: 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh [hunk 3]
```diff
@@ -329,8 +329,7 @@ EOF
 #   None
 #########################
 ldap_add_custom_ldifs() {
-    info "Loading custom LDIF files..."
-    warn "Ignoring LDAP_USERS, LDAP_PASSWORDS, LDAP_USER_DC and LDAP_GROUP environment variables..."
+    info "Loading custom LDIF files... (extra schemas won't be added, default tree won't be created)"
```
Comment (andresbono) [inline 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:332]: I'm not sure if this message belongs to this function. The fact that "extra schemas won't be added, default tree won't be created" depends on the if else block where `ldap_add_custom_ldifs` is called, not to this function itself...

**Comment by andresbono** [comment]
> Thanks for the review! I've changed the branch (as the patch is small, I amended the commit to avoid cluttering the history).
> 
> I'm now proposing to promote "default tree" as a vocabulary term (think "set expression") in the documentation and code. Thus I've added a small section in the README to explain what this default tree is about (moving some info from the configuration section). I think this makes things a bit clearer.

💯  Thank you very much!

**Pull request closed without merging**